docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,861 @@
+      "mask": 200
+    },
+    "apply_actions": [
+      "SET_FIELD"
+    ],
+    "write_actions": [
+      "SET_FIELD"
+    ]
+  },
+  "expected": {
+    "table_id": 10,
+    "priority": 5,
+    "flags": [],
+    "match": [
+      {
+        "value": 9,
+        "mask": 10,
+        "field": "IN_PORT"
+      }
+    ],
+    "instructions": {
+      "go_to_table": 3,
+      "go_to_meter": 4,
+      "write_metadata": {
+        "value": 100,
+        "mask": 100
+      },
+      "apply_actions": [
+        "SET_FIELD"
+      ],
+      "write_actions": [
+        "SET_FIELD"
+      ]
+    }
+  },
+  "actual": {
+    "table_id": 1,
+    "priority": 2,
+    "flags": [
+      "RESET_COUNTERS"
+    ],
+    "match": [
+      {
+        "value": 7,
+        "mask": 8,
+        "field": "ETH_TYPE"
+      }
+    ],
+    "instructions": {
+      "go_to_table": 1,
+      "go_to_meter": 1,
+      "write_metadata": {
+        "value": 10,
+        "mask": 10
+      },
+      "apply_actions": [
+        "PORT_OUT"
+      ],
+      "write_actions": [
+        "PORT_OUT"
+      ]
+    }
+  }
+}
+```
+
+### Validate rules and meters installed on the switch
+
+`GET https://{host}/v2/switches/{switch_id}/validate`
+
+**Default: without any query params response contains extended info about groups, lags, meters and
+rules.**
+
+Response payload(`extended info for all fields`)
+
+```json
+{
+  "groups": {
+    "excess": [
+      <extendedGroupInfo>
+    ],
+    "misconfigured": [
+      <extendedGroupInfo>
+    ],
+    "missing": [
+      <extendedGroupInfo>
+    ],
+    "proper": [
+      <extendedGroupInfo>
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "excess": [
+      <extendedLogicalPortInfo>
+    ],
+    "misconfigured": [
+      <extendedLogicalPortInfo>
+    ],
+    "missing": [
+      <extendedLogicalPortInfo>
+    ],
+    "proper": [
+      <extendedLogicalPortInfo>
+    ]
+  },
+  "meters": {
+    "excess": [
+      <extendedMeterInfo>
+    ],
+    "misconfigured": [
+      <extendedMeterInfo>
+    ],
+    "missing": [
+      <extendedMeterInfo>
+    ],
+    "proper": [
+      <extendedMeterInfo>
+    ]
+  },
+  "rules": {
+    "excess": [
+      <extendedRuleInfo>
+    ],
+    "misconfigured": [
+      <extendedRuleInfo>
+    ],
+    "missing": [
+      <extendedRuleInfo>
+    ],
+    "proper": [
+      <extendedRuleInfo>
+    ]
+  }
+}
+```
+
+### Including specific info in response
+
+### Request params
+
+To get only group info in response use query param `GROUP`, other fields will be omitted.
+
+`GET https://{host}/v2/switches/{switch_id}/validate?include=GROUP`
+
+Response payload
+
+```json
+{
+  "groups": {
+    "excess": [
+      <extendedGroupInfo>
+    ],
+    "misconfigured": [
+      <extendedGroupInfo>
+    ],
+    "missing": [
+      <extendedGroupInfo>
+    ],
+    "proper": [
+      <extendedGroupInfo>
+    ]
+  }
+}
+```
+
+To get only meters info in response use query param `METER`, other fields will be omitted.
```

Comment:
Too detailed explanation. Just say - that all other targets can be selected in same way i.e `include=meters` for meters, `include=rules` and so on. 

PS We need to use field response name into include/exclude arguments or define a mapping between used values into include/exclude arguments and result fields.